pyvista/core/pointset.py

```diff
@@ -339,7 +339,134 @@ def rotate_vector(self, *args, **kwargs):
+    geometry. This class is useful for improving the performance of filters on
+    point clouds, but not plotting.
+
+    For further details see `VTK: vtkPointSet Details
+    <https://vtk.org/doc/nightly/html/classvtkPointSet.html#details>`_
+
+    Parameters
+    ----------
+    points : Sequence, optional
+        List, numpy array, or sequence containing point locations. Must be an
+        ``(N, 3)`` array of points.
+
+    editable : bool, optional
+        Set whether this dataset is editable after creation. Default
+        ``True``. For performance, set this to ``False``.
+
+    deep : bool, optional
+        Whether to copy the input ``points``, or to create a PointSet from them
+        without copying them.  Setting ``deep=True`` ensures that the original
+        arrays can be modified outside the mesh without affecting the
+        mesh. Default is ``False``.
+
+    force_float : bool, optional
+        Casts the datatype to ``float32`` if points datatype is non-float.
+        Default ``True``. Set this to ``False`` to allow non-float types,
+        though this may lead to truncation of intermediate floats when
+        transforming datasets.
+
+    Notes
+    -----
+    This class requires ``vtk>=9.1.0``. This is an abstract class in
+    ``vtk<9.1.0`` and cannot be instantiated.
+
+    Examples
+    --------
+    Create a simple point cloud of 10 points from a numpy array.
+
+    >>> import numpy as np
+    >>> import pyvista
+    >>> rng = np.random.default_rng(0)
+    >>> points = rng.random((10, 3))
+    >>> pset = pyvista.PointSet(points)
+
+    Plot the pointset. Note: this casts to a ``pyvista.PolyData`` internally
+    when plotting.
+
+    >>> pset.plot(point_size=10)
+
+    """
+
+    def __init__(self, points=None, editable=True, deep=False, force_float=True):
+        """Initialize the pointset."""
+        if pyvista.vtk_version_info < (9, 1, 0):
+            raise VTKVersionError("pyvista.PointSet requires VTK >= 9.1.0")
+
+        super().__init__()
+        if points is not None:
+            self.SetPoints(pyvista.vtk_points(points, deep=deep, force_float=force_float))
+
+        self.editable = editable
+
+    def __repr__(self):
+        """Return the standard representation."""
+        return DataSet.__repr__(self)
+
+    def __str__(self):
+        """Return the standard str representation."""
+        return DataSet.__str__(self)
+
+    @property
+    def editable(self) -> bool:
+        """Return or set if this dataset be incrementally modified.
+
+        Examples
+        --------
+        >>> import numpy as np
+        >>> import pyvista
+        >>> rng = np.random.default_rng(0)
+        >>> points = rng.random((10, 3))
+        >>> pset = pyvista.PointSet(points, editable=False)
+        >>> pset.editable
+        False
+
+        Allow the pointset to be editable.
+
+        >>> pset.editable = True
+        >>> pset.points[:] = 0
```

Comment:
I have multiple issues with editability here:
  1. It doesn't look like editability works right now. If I keep `pset.editable = False` then I can still do `pset.points = ...` or `pset.points[:] = ...` or `pset.SetPoints(pyvista.vtk_points(...))` and they all seem to work, and they all seem to update the points. Despite `editable = False`.
  2. Even if editability would work and prevent `pset.points = ...`, I would expect that `pset.points[:] = 0` would still work. Because what this should do if I understand correctly is `tmp = pset.points; tmp[:] = 0`. VTK has no idea that we pulled out a reference to the points array with a `pyvista_ndarray` of ours, and it has no chance to prevent the mutation of said array.

So I think the `editable` behaviour needs checking, and the doctest example should probably use non-sliced `pset.points = 0` to show its use. Even better would be an example that errors when `editable=False` is passed. And if I'm right about this then we might consider making this an explicit point in the docs, because it's subtle behaviour (although I would expect most end users to use regular assignment without slicing).

Reply:
Editable is still on the list, and I'm planning on enforcing it with:
```py
>>> import numpy as np
>>> array = np.ones((10, 3))
>>> array.flags.writeable = False
>>> array[:] = 0
ValueError: assignment destination is read-only
```

Reply:
OK, current implementation looks good, thanks.